proposal/color-4-new-spaces-js.d.ts.md

```diff
@@ -0,0 +1,931 @@
+
+These will replace the [existing constructors] for legacy colors.
+
+#### HSL Constructor
+
+Create a new SassColor in the `hsl` color space.
+
+* If `options.alpha` is `null` and `options.space` is not set, emit a
+  deprecation warning named `null-alpha`.
+  
+* Let `hue` be the result of [parsing a channel value] with value `options.hue`.
+
+* Let `saturation` be the result of [parsing a channel value] with value `options.saturation`.
+
+* Let `lightness` be the result of [parsing a channel value] with value `options.lightness`.
+  
+* If `options.alpha` is not set, let `alpha` be `1`. Otherwise, let `alpha` be
+    the result of [parsing a channel value] with value `options.alpha`.
+
+* Set [`internal`] to the result of [`hsl(hue saturation lightness / alpha)`].
+
+```ts
+constructor(options: {
+  hue: number | null;
+  saturation: number | null;
+  lightness: number | null;
+  alpha?: number | null;
+  space?: 'hsl';
+});
+```
+
+#### HWB Constructor
+
+Create a new SassColor in the `hwb` color space.
+
+* If `options.alpha` is `null` and `options.space` is not set, emit a
+  deprecation warning named `null-alpha`.
+
+* Let `hue` be the result of [parsing a channel value] with value `options.hue`.
+
+* Let `whiteness` be the result of [parsing a channel value] with value `options.whiteness`.
+
+* Let `blackness` be the result of [parsing a channel value] with value `options.blackness`.
+  
+* If `options.alpha` is not set, let `alpha` be `1`. Otherwise, let `alpha` be
+    the result of [parsing a channel value] with value `options.alpha`.
+
+* Set [`internal`] to the result of [`hwb(hue whiteness blackness / alpha)`].
+
+```ts
+constructor(options: {
+  hue: number | null;
+  whiteness: number | null;
+  blackness: number | null;
+  alpha?: number | null;
+  space?: 'hwb';
+});
+```
+
+#### RGB Constructor
+
+Create a new SassColor in the `rgb` color space.
+
+* If `options.alpha` is `null` and `options.space` is not set, emit a
+  deprecation warning named `null-alpha`.
+
+* Let `red` be the result of [parsing a channel value] with value `options.red`.
+
+* Let `green` be the result of [parsing a channel value] with value
+    `options.green`.
+
+* Let `blue` be the result of [parsing a channel value] with value
+    `options.blue`.
+
+* If `options.alpha` is not set, let `alpha` be `1`. Otherwise, let `alpha` be
+      the result of [parsing a channel value] with value `options.alpha`.
+
+* Return the result of `rgb(red green blue / alpha)
+
+```ts
+constructor(options: {
+  red: number | null;
+  green: number | null;
+  blue: number | null;
+  alpha?: number | null;
+  space?: 'rgb';
+});
+```
+
+```ts
+}
+```
+
+### Deprecations
+
+A number of SassColor getters only make sense for [legacy color space], and so
+are being deprecated for `channel`. This deprecation is called `color-4-api`.
+
+* `red`
+* `green`
+* `blue`
+* `hue`
+* `saturation`
+* `lightness`
+* `whiteness`
+* `blackness`
+* `alpha`
```

Comment:
@jamesnw Is `alpha` being deprecated? It looks like it's identical to the "new" `alpha()` fn [described above](https://github.com/sass/sass/blob/main/proposal/color-4-new-spaces-js.d.ts.md#alpha).

Reply:
Good catch, `alpha` is not deprecated. Updated in https://github.com/sass/sass/pull/3704/commits/b9484d122369877c39e3edc6a3219011afbb6c5f